qualtran/resource_counting/_bloq_counts.py

```diff
@@ -128,8 +139,38 @@ class GateCounts:
     cswap: SymbolicInt = 0
     and_bloq: SymbolicInt = 0
     clifford: SymbolicInt = 0
-    rotation: SymbolicInt = 0
     measurement: SymbolicInt = 0
+    binned_rotation_epsilons: Counter[FloatRepr_T] = field(
+        factory=Counter, converter=_mapping_to_counter, eq=lambda d: tuple(d.items())
+    )
+
+    @classmethod
+    def from_rotation_with_eps(
+        cls, eps: SymbolicFloat, *, n_rotations: int = 1, eps_repr_prec: int = 10
+    ):
+        """Construct a GateCount with a rotation of precision `eps`.
+
+        Formats the value of `eps` as a string using `np.format_float_scientific`,
+        to use as a safe dictionary key. If `eps` is symbolic, it is used as-is.
+
+        Args:
+            eps: precision to synthesize the rotation(s).
+            eps_repr_prec: number of digits to approximate `eps` to. Uses 10 by default.
+                           See `np.format_float_scientific` for more details.
+                           If `eps` is symbolic, this parameter is ignored.
+            n_rotations: number of rotations, defaults to 1.
+        """
+        if is_symbolic(eps):
+            eps_bin: FloatRepr_T = eps
+        else:
+            eps_bin = np.format_float_scientific(eps, precision=eps_repr_prec, unique=False)
```

Comment:
why `unique=False`?

Reply:
This was to ensure that the representation was more stable. from the docs:
> [... If precision is given fewer digits than necessary can be printed. ...](https://numpy.org/doc/stable/reference/generated/numpy.format_float_scientific.html#:~:text=If%20precision%20is%20given%20fewer%20digits%20than%20necessary%20can%20be%20printed.)

Reply:
Actually, I'm not sure what the best way is to check if two rotation frequency tables are the same, as the epsilons could have been stored with different precisions, or be approximate.